### DISCUSSION TITLE
Paperless-ng is keeping my unraid server occupied

### DISCUSSION THREAD

**Discussion by Miesjelangelo** [jonaswinkler/paperless-ng#513] [Support] [answered]
Hi there,

Thanks for making this software available first of all. Love it! 

I have it running in a docker container on unraid. Everything seems to be working as expected (all functions are working i believe). But when the docker is running, my disk keeps busy even while i'm not using paperless. It might have to do something with the loop for consuming new docs uploaded to the folder its monitoring. 

Is there a way to change this interval to lets see every hour or so? 

Any other suggestions are more than welcome ofcourse :)

**Comment by jonaswinkler** [comment]
By default, paperless uses file system notifications to detect new files in the consumption folder, unless `PAPERLESS_CONSUMER_POLLING=x` is specified. If it is, paperless polls the consumption directory every `x` seconds.

You could try setting this to a high value, such as `PAPERLESS_CONSUMER_POLLING=600` for every 10 minutes and see if that's the cause for your issues.

**Reply by Miesjelangelo** [reply]
Thanks for the very quick response! Found and changed the setting you mentioned, but it keeps my unraid server busy. Every second, my disk spins. 

I see that the CPU spikes in the same pattern as well. As soon as i turn of the paperless docker, both stay quiet. Any other suggestions perhaps?

**Reply by jonaswinkler** [reply]
Not sure. These are guesses without doing any investigation.

* Could be redis doing some regular background work. Or postgresql. What if you stop paperless, but keep these two running?
* Any messages in the logs that occur at the same frequency?

Apart from the configured schedules (and there aren't any that run every second), I don't do any recurring work in paperless.

I'd use some disk IO analysis tools in the container to see what files are getting read / written to. That might help to narrow down the issue. I know there are some tools to do this, but I can't remember the names right now.

**Reply by jonaswinkler** [reply]
I did some changes to how redis is used in 1.1.0. It's now not only used for communicating tasks between different processes of paperless, but for caching and WebSocket messages as well;

Might be worth checking if this also happens on 1.0.0 (no migrations needed, switchin between these two is safe).
* If it does not, some configuration issue with the new features.
* If it does, some general configuration issue with redis.

**Reply by Miesjelangelo** [reply]
IT does not help. I'll dig a little deeper in the redis configuration.

**Reply by Miesjelangelo** [reply]
Looks like i fixed it!

I'll explain what I did for others, but I believe its interesting for users of Unraid only. 

Before trroubleshooting, I installed Redis via the Community apps platform. I picked the one from Bitnami. I did all configuration, with and without password, and it all worked as expected. But since I don't like the fact that my harddrive is constantly active, i started looking for a solution. 

Since I, with the help of @jonaswinkler , figured out that the issue was happening somehwere between Paperless and Redis, I removed the Bitnami Redis and installed the Redis docker container manually via the Add container button (used redis:latest to make sure I recieved the latest version). I configured the port, removed the password line from the paperless docker settings and rebooted both dockers. After that, uploading works and the disk stops working after it's done with its tasks. 

@jonaswinkler , would you recommend using a password between Redis and Paperless-ng?

**Reply by jonaswinkler** [reply]
Unless you expect someone to connect to redis and read entries from paperless for malicous purposes or post bogus messages to the task queue, I don't think a password is strictly necessary, but I'll look into it.

**Comment by Miesjelangelo** [comment, marked as answer]
Looks like i fixed it!

I'll explain what I did for others, but I believe its interesting for users of Unraid only. 

Before trroubleshooting, I installed Redis via the Community apps platform. I picked the one from Bitnami. I did all configuration, with and without password, and it all worked as expected. But since I don't like the fact that my harddrive is constantly active, i started looking for a solution. 

Since I, with the help of @jonaswinkler , figured out that the issue was happening somehwere between Paperless and Redis, I removed the Bitnami Redis and installed the Redis docker container manually via the Add container button (used redis:latest to make sure I recieved the latest version). I configured the port, removed the password line from the paperless docker settings and rebooted both dockers. After that, uploading works and the disk stops working after it's done with its tasks. 

@jonaswinkler , would you recommend using a password between Redis and Paperless-ng?
